core/include/detray/navigation/navigator.hpp

```diff
@@ -573,25 +575,25 @@ class navigator {
         /// Can never be advanced beyond the last element
         dist_t m_last{-1};
 
-        /// The inspector type of this navigation engine
-        inspector_type m_inspector;
-
         /// The navigation status
         navigation::status m_status = navigation::status::e_unknown;
```

Comment:
Should we put these at the back of the struct for packing reasons?

Reply:
Indeed, I tried shuffling things around to avoid padding. The types in the ```navigation``` namespace should have only one byte now (I measured this as well), the same as the booleans, so I tried ordering them by access probability (by guessing so far :)).  The inspector *should* have no size at all (in production code it is the void inspector). But there are still 7 bytes popping up somewhere that I cannot account for, even if I remove the inspector completely. I guess, I will have a look at the member offsets now to see where it is

Reply:
FYI you may need to mark the inspector as having no unique address: https://en.cppreference.com/w/cpp/language/attributes/no_unique_address

Reply:
Another good point, but I already tried that :) (see commented code below). Maybe those bytes come from the base class?

Reply:
I checked the member offsets, there is no padding between them and the inspector state does not take any memory. I think that the compiler is adding extra padding to align the entire struct to 8 bytes though.

Reply:
Removed one byte from the state to have it align to 8 bytes naturally and the mystery bytes indeed vanished